Gnokey grant (#16)
gnolang/ecosystem-fund-grants: grants/gnokey/gnokey.md

# Gno Grant#3 Gnokey

# Berty X Gno Grant #3 Application

- **Name:** Berty Technologies
- **Project name**: Gnokey

> This document is only a DRAFT Grant application and it describes the idea behind the Gnokey application. It purpose is to serve as support to polish the collaborative approach we are proposing in this grant. 

Reviewer: "It purpose" should be "Its purpose"

> This document is only a DRAFT Grant application and it describes the idea behind the Gnokey application. It purpose is to serve as support to polish the collaborative approach we are proposing in this grant. 

Reviewer: ```suggestion
This document is only a DRAFT Grant application and it describes the idea behind the Gnokey application. Its purpose is to serve as support to polish the collaborative approach we are proposing in this grant. 
```

> In a nuthsell the Gnokey application will be a mobile application that users will use to SECURELY manage their key and signt the transactions. One of the main benefits is that (D)apps builders will not have to worry about building in their apps the security part and they could just “outsource” it to the Gnokey app and remain focused on building their app’s main functionalities.

Reviewer: "signt" should be "sign"

> In a nuthsell the Gnokey application will be a mobile application that users will use to SECURELY manage their key and signt the transactions. One of the main benefits is that (D)apps builders will not have to worry about building in their apps the security part and they could just “outsource” it to the Gnokey app and remain focused on building their app’s main functionalities.

Reviewer: "manage their key" should be "manage their keys"

> In a nuthsell the Gnokey application will be a mobile application that users will use to SECURELY manage their key and signt the transactions. One of the main benefits is that (D)apps builders will not have to worry about building in their apps the security part and they could just “outsource” it to the Gnokey app and remain focused on building their app’s main functionalities.

Reviewer: ```suggestion
In a nuthsell the Gnokey application will be a mobile application that users will use to SECURELY manage their keys and sign the transactions. One of the main benefits is that (D)apps builders will not have to worry about building in their apps the security part and they could just “outsource” it to the Gnokey app and remain focused on building their app’s main functionalities.
```

To build Gnokey we at Berty are proposing a 3 step approach: 

- Step 1: identify main features of Gnokey as well as a high-level architecture or system diagram - DONE
- Step 2: work with Gno team to identify a 3rd party security expert that can help detail the technical architecture so we make sure that Gnokey is secure and that users can trust it. This would actually be the objective of the Grant proposal: to obtain a technical implementation plan that we can later implement.

> - Step 3: building on the technical implementation plan from Step 2 (Gnokey Grant), we decide who is best positioned to build the app. I[t](http://app.It) could be the Berty team, Gno or someone else.

Reviewer: "I[t](http://app.It)" should be "It"

> - Step 3: building on the technical implementation plan from Step 2 (Gnokey Grant), we decide who is best positioned to build the app. I[t](http://app.It) could be the Berty team, Gno or someone else.

Reviewer: ```suggestion
- Step 3: building on the technical implementation plan from Step 2 (Gnokey Grant), we decide who is best positioned to build the app. It could be the Berty team, Gno or someone else.
```

For the moment we are looking for feedback on the approach for Step 2 - which is actually this grant proposal.

FEEDBACK NEEDED: please have a look at the videos below that describe our proposed approach as well as an explanations of the high-level technical diagram:

- General Approach: [https://www.loom.com/share/27b0aaf07d2c4141bbf753efa5b2e189](https://www.loom.com/share/27b0aaf07d2c4141bbf753efa5b2e189)

> - Tech explanations of the HL architecture (view in desktop browser or desktop Miro app): [https://miro.com/app/board/uXjVN3d9Pts=/?playRecording=3439faf7-4413-4ebb-9d36-51aac2be588b](https://miro.com/app/board/uXjVN3d9Pts=/?playRecording=3439faf7-4413-4ebb-9d36-51aac2be588b)

Reviewer: I suggest to add "and click play". (It may not be obvious in the Miro screen.)

> - Tech explanations of the HL architecture (view in desktop browser or desktop Miro app): [https://miro.com/app/board/uXjVN3d9Pts=/?playRecording=3439faf7-4413-4ebb-9d36-51aac2be588b](https://miro.com/app/board/uXjVN3d9Pts=/?playRecording=3439faf7-4413-4ebb-9d36-51aac2be588b)

Reviewer: ```suggestion
- Tech explanations of the HL architecture (view in desktop browser or desktop Miro app and click play): [https://miro.com/app/board/uXjVN3d9Pts=/?playRecording=3439faf7-4413-4ebb-9d36-51aac2be588b](https://miro.com/app/board/uXjVN3d9Pts=/?playRecording=3439faf7-4413-4ebb-9d36-51aac2be588b)
```
